zilliqa/src/contracts/intershard_bridge.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity ^0.8.4;
+
+contract IntershardBridge {
+    event Relayed(
+        uint64 indexed targetChainId,
+        address source,
+        bool contract_creation,
+        address target,
+        bytes call,
+        uint64 gasLimit,
+        uint128 gasPrice,
+        uint64 indexed nonce
+    );
+
+    // shard_id => nonce => is_dispatched
+    mapping(uint64 => mapping(uint64 => bool)) public dispatched;
```

Comment:
They're meant to be ascending but not necessarily sequential, and mirrored across both sides of the bridge.

But as mentioned in the call today, we don't actually need the nonces because the transactions will only come from a local (light) node that we trust, so we just need to make sure our node/lightnode architecture doesn't double-notify us of cross-shard events. On the source shard you obviously have standard transaction replay protection, so such a double-notification would only happen due to some major error or bug in the light node.

Reply:
If we'll remove the nonces then lets leave it like this then :) I was thinking if `mapping(uint64 => uint64[])` would make more sense, but its not important.